odbc/src/api/functions.rs

```diff
@@ -337,13 +337,82 @@ fn sql_alloc_handle(
 #[no_mangle]
 pub unsafe extern "C" fn SQLBindCol(
     hstmt: HStmt,
-    _col_number: USmallInt,
-    _target_type: SmallInt,
-    _target_value: Pointer,
-    _buffer_length: Len,
-    _length_or_indicatior: *mut Len,
+    col_number: USmallInt,
+    target_type: SmallInt,
+    target_value: Pointer,
+    buffer_length: Len,
+    length_or_indicator: *mut Len,
 ) -> SqlReturn {
-    unimpl!(hstmt);
+    panic_safe_exec_keep_diagnostics!(
+        debug,
+        || {
+            let mongo_handle = MongoHandleRef::from(hstmt);
+            let stmt = must_be_valid!((*mongo_handle).as_statement());
+
+            // Currently, we only support column binding of one row at a time.
+            if stmt.attributes.read().unwrap().row_array_size != 1
```

Comment:
uhm, yes, from the SSIS traces, the tool is binding a rowset, not a single row.
Also, if we want to start supporting only one row at a time, we need to add the protection on the SQLSetStmtAttribute, not just on SQLBind. It's good to have it, but it is better to fail earlier so that it is clear we don't support rowset binding (yet)

Reply:
I'll add the checks to `SQLSetStmtAttrW` in this pr.

Reply:
I was adding checks to `SQLSetStmtAttrW`, but I'm not sure what to do about `SQL_ATTR_ROW_ARRAY_SIZE` because it modifies the same field as `SQL_ROWSET_SIZE`. Do we need to add another field to `StatementAttributes` and break these 2 up? I think we should add another field called `rowset_size` or something like that.

```
        StatementAttribute::SQL_ATTR_ROW_ARRAY_SIZE | StatementAttribute::SQL_ROWSET_SIZE => {
            match FromPrimitive::from_i32(value_ptr as i32) {
                Some(ras) => {
                    stmt.attributes.write().unwrap().row_array_size = ras;
                    SqlReturn::SUCCESS
                }
                None => {
                    stmt_handle
                        .add_diag_info(ODBCError::InvalidAttrValue("SQL_ATTR_ROW_ARRAY_SIZE"));
                    SqlReturn::ERROR
                }
            }
        }
```

Reply:
I think both have the same meaning.
SQLExtendedFetch uses SQL_ROWSET_SIZE and SQLFetchScroll uses SQL_ATTR_ROW_ARRAY_SIZE, but SQLFetchScroll is essentially the ODBC 3.x version of SQLExtendedFetch.
https://learn.microsoft.com/en-us/sql/odbc/reference/syntax/sqlextendedfetch-function?view=sql-server-ver16

It seems fine to keep it as is. We don't support SQLExtendedFetch and if we were to I think we would be Ok with the same internal variable for the row set size. 
A note on that : The signature is not in functions.rs and that could be potentially be an issue when the DM tries to pass the call to the driver.